pkg/startup/startup.go

```diff
@@ -1113,8 +1119,19 @@ func setNodeNetworkUnavailableFalse(config *rest.Config, nodeName string) error
 		return err
 	}
 	patch := []byte(fmt.Sprintf(`{"status":{"conditions":%s}}`, raw))
-	_, err = clientset.CoreV1().Nodes().PatchStatus(nodeName, patch)
-	return err
+	to := time.After(30 * time.Second)
+	for {
+		select {
+		case <-to:
+			err = fmt.Errorf("timed out patching node, last error was: %s", err.Error())
+			return err
+		default:
+			_, err = clientset.CoreV1().Nodes().PatchStatus(nodeName, patch)
+			if err != nil {
+				log.WithError(err).Warnf("Failed to set NetworkUnavailable to False; will retry")
+			}
```

Comment:
I think we should break the loop here if we're successful, right?

Reply:
Yes! (and, duh!, my bad.)